R/utils.R

```diff
@@ -4,37 +4,45 @@ setup_future <- function(jobs) {
     ## If running as a script enable this
     options(future.fork.enable = TRUE)
   }
-  
-  
+
+
   future::plan("multiprocess", workers = min(future::availableCores(), jobs),
                gc = TRUE, earlySignal = TRUE)
-  
-  
+  futile.logger::flog.debug("Checking the cors available - %s cores and %s jobs. using %s workers",
+                            future::availableCores(),
+                            jobs,
+                            min(future::availableCores(), jobs))
+
   jobs <- max(1, ceiling(future::availableCores() / jobs))
   return(jobs)
 }
 
 
 #' Check data to see if updated since last run
 check_for_update <- function(cases, last_run) {
-  
+
   current_max_date <- max(cases$date, na.rm = TRUE)
-    
-  if (file.exists(last_run)){
+
+  if (file.exists(last_run)) {
+    futile.logger::flog.trace("last_run file (%s) exists, loading.", last_run)
     last_run_date <- readRDS(last_run)
-   
+
     if (current_max_date <= last_run_date) {
-      stop("Data has not been updated since last run. 
-      If wanting to run again then remove ", last_run)
+      futile.logger::flog.info("Data has not been updated since last run. If wanting to run again then remove %s", last_run)
+      futile.logger::flog.debug("Max date in data - %s, last run date from file - %s",
+                                format(current_max_date, "%Y-%m-%d"),
+                                format(last_run_date, "%Y-%m-%d"))
+      return(FALSE)
     }
   }
-    
+  futile.logger::flog.debug("New data to process")
   saveRDS(current_max_date, last_run)
-  
-  return(invisible(NULL))
+
+  return(TRUE)
 }
 #' Clean regional data
 clean_regional_data <- function(cases) {
```

Comment:
Something for the future that I will mention here is that it would be good to be able to run on different lengths of time. The default we are using here is a sliding window of 12 weeks but at some point, it would be really nice to have two phases of updates. So one every day looking at part of the time series and one less often that runs on all the data (or adds it together piecewise if we can implement that at a later date). We currently have no way of presenting that on the website but once the estimates are in place it could be more easily thought about.